src/node_buffer.cc

```diff
@@ -406,7 +406,17 @@ MaybeLocal<Object> New(Environment* env,
     return Local<Object>();
   }
 
-  Local<ArrayBuffer> ab = ArrayBuffer::New(env->isolate(), data, length);
+
+  std::unique_ptr<BackingStore> backing =
+      ArrayBuffer::NewBackingStore(data,
+                                   length,
+                                   [](void*, size_t, void*){},
+                                   nullptr);
+  Local<ArrayBuffer> ab = ArrayBuffer::New(env->isolate(),
+                                           std::move(backing));
+  // TODO(thangktran): drop this check when V8 is pumped to 8.0 .
+  if (!ab->IsExternal())
+    ab->Externalize(ab->GetBackingStore());
```

Comment:
Thanks, this seems like one of the trickier ones but I like your solution. :+1:

It might look a bit odd to readers that the deletion callback above is empty; maybe it makes sense to add a comment saying that `CallbackInfo` is used to release the data?

Reply:
fixed